Emit deprecation warnings when boosts are defined in mappings

romseygeek opened the pull request:
We removed index-time boosting back in 5x, and we no longer document the 'boost'
parameter on any of our mapping types.  However, it is still possible to define an
index-time boost on a field mapper for a surprisingly large number of field types, and
they even have an effect (sometimes, on some queries).

As a first step in finally removing all traces of index time boosting, this comment emits
a deprecation warning whenever a boost parameter is found on a mapping definition.

romseygeek committed:
Emit deprecation warnings when boosts are defined in mappings

elasticmachine commented:
Pinging @elastic/es-search (:Search/Mapping)

romseygeek commented:
The next step will be to throw an error on indexes created in 8x, and remove MappingFieldType#boost entirely.

jimczi commented on server/src/main/java/org/elasticsearch/index/mapper/ParametrizedFieldMapper.java:
```diff
@@ -570,6 +570,9 @@ public final void parse(String name, ParserContext parserContext, Map<String, Ob
                         + "] on mapper [" + name + "] of type [" + type + "]");
                 }
+                if (Objects.equals("boost", propName)) {
+                    deprecationLogger.deprecate("boost", "Parameter [boost] on field [{}] is deprecated", name);
```
maybe add that it will be removed in 8 since that's the new plan ?

romseygeek replied:
++

jimczi reviewed (commented):
I wonder if it's possible to warn only when the field or index is created ? Not really an issue but that could avoid duplicate log messages every time there is an update on the mapping ? Maybe that's a feature though and having lots of deprecation messages is a good thing.
Anyway, if the plan is to remove this option entirely in 8, +100 to deprecate again in 7x in order to raise the awareness again. Maybe also add a note in the deprecation page, it's already there in 5x but the new plan is to remove in 8 so there should be some warnings in the release notes as well ?

romseygeek committed:
deprecation notice; update warnings to include version

romseygeek commented:
I've added version information to the deprecation warning messages.

> Maybe also add a note in the deprecation page, it's already there in 5x but the new plan is to remove in 8 so there should be some warnings in the release notes as well ?

I've added a note to the `migrate_8_0/mappings` doc, and I'll move it to the `migrate_7_10` doc on backport.

romseygeek committed:
duh

jimczi reviewed (approved):
LGTM

romseygeek committed:
cmon
